sed/binning.py

```diff
@@ -36,6 +35,7 @@ def _simplify_binning_arguments(
     axes: Union[str, Sequence[str]] = None,
     ranges: Sequence[Tuple[float, float]] = None,
 ) -> tuple:
+    """TODO needs docstring"""
```

Comment:
@steinnymir I don't undestand what this function is supposed to do, how it works, and why it is called both in bin_dataframe and bin_partition. Therefore I don't know where to reasonably place the extension from bin centers to bin edges. Please provide a docstring and explanation, and tests for all cases.

Reply:
This allows to mimic the input arguments of numpy.histogramdd to the bin_dataframe and bin_partition functions.

I externalized this in a function to remove the overhead of running through this at each parallel iteration of bin_partition. I'll write the docstring for this, and tests as they might be very important here.

Reply:
I understand the general purpose, but not the exact function. Also, I am wondering why you call it both in bin_dataframe and bin_partition. Would it not be cleaner to restrict the possible inputs to bin_partition to those of np.histogram, and only call it in bin_dataframe?

Reply:
I don't see the use case for using bin_partition standalone outside a test case, where one can use restricted input definition.

Reply:
It is called in both as bin_partition could be used standalone. I did in fact use it when using pandas dataframes, and not dask ones.

I agree that this function looks horrible like this, and if there is a more elegant way to check and transform inputs, I'm happy to change this. This was the way I found so far to ensure the same signature (almost) for `np.histogramdd`, `bin_partition` and `bin_dataframe`...

Reply:
I see. The reason I argue to call it only once is that I was wondering where to extend input bin-lists from bin-centers into bin-eges as we discussed. A natural place seemed for me this function, but this only works if it only is called once. I don't know how to handle this and where to place this call, if bin_partition can be called standalone, and should have the same functionalitiy. This would make things much more complicated here.

Reply:
This is indeed called only once. In bin_dataframe, all calls to bin_partition are passing the flag skipTest, which makes bin_partition ignore this function, and assume the input is already correct.

Reply:
I see. Regarding the points for the representation in the xarray, I still think this is a separate issue that we should address in a separate branch. I realized this is also not correct the way we have done it so far for the case of num_bins and ranges, because the bins edges returned by np.histdd enclose the given range, whereas the bin centers we returned included the same range. We need to discuss how to handle this, and what the user expected behavior should be.

Reply:
Please correct me if I'm wrong. Up to now, this `bin_dataframe` function returns bins in the same way as np.histogramdd does. If that's the case, IMO that should be the default.
If for some reason, you guys think there should be another way of returning bins, we could think of passing a flag to the function to return this non-defaults

Reply:
I agree this is still an open question. This PR needs to go though, so I will merge this and we can start new PRs from the issue you created and making other ones.